src/agencia/Agencia.ts

```diff
@@ -0,0 +1,60 @@
+// src/agencia/Agencia.ts
+import { Cliente } from "../cliente/Cliente";
+
+export interface IAgencia {
+  getId(): string;
+  getNome(): string;
+  getEndereco(): string;
+  getClientes(): Cliente[];
+  adicionarCliente(cliente: Cliente): void;
+  listarClientes(): Cliente[];
+  removerCliente(clienteId: string): boolean;
+}
```

Comment:
Lu, uma abordagem que tem me ajudado bastante na organização dos arquivos do projeto é o uso do "Modelo em Camadas". Embora possa não ser a única abordagem possível, ela fornece uma direção clara sobre as responsabilidades de cada elemento da nossa aplicação.

Se optar por seguir essa abordagem, a estrutura do projeto ficaria algo assim:

```shell
src/
|-- modelos/
|   |-- Cliente.ts
|   |-- Agencia.ts
|   |-- Contas.ts
|
|-- interfaces/
|   |-- IAgencia.ts
|   |-- IContas.ts
|
|-- servicos/
|   |-- Emprestimo.ts
|
```

Gosto dessa divisão porque, conforme vamos implementando as coisas, os arquivos não ficam tão grandes. Além disso, é um padrão comum, então quem for olhar o código depois consegue entender melhor o fluxo e a responsabilidade de cada classe.

Mas, claro, fica a teu critério decidir qual a melhor estrutura pro projeto.

Reply:
Boa, obrigada, eu havia separado as interfaces em cada pasta mas não consegui puxar os métodos privates, eles davam erro cada hora em uma pasta, dai fiz um arquivo. mas não tinha pensado assim.

Reply:
Muda pra `protect`. Assim você consegue compartilhar com as classes que, na teoria, herdam esse método.

Reply:
O `private` você só consegue usar o método dentro da própria classe, impossibilitando o acesso as classes filhas.

Reply:
> Muda pra `protect`. Assim você consegue compartilhar com as aulas que, na teoria, herdam esse método.

certo, vou fazer isso, obrigada!

Reply:
Coloquei as interfaces separadas dentro de cada pasta e coloquei o passo a passo no Readme, obrigada novamente Lu! ❤️